### DISCUSSION TITLE
How to search with conditions

### DISCUSSION THREAD

**Discussion by anoonoll** [algolia/instantsearch#5296] [Q&A] [answered]
I want to do a search search with conditions.
I tried to search Doc, but I couldn't find a way to search with conditions in React.
I can see how to do it in PHP, Ruby and Javascript, but not in React.
Ideally, I'd like to do a search that matches the selected values, as in the link below.
If you know of a good way to do this, I would appreciate it if you could let me know.
https://react-instantsearch.netlify.app/storybook/?path=/story/dynamicwidgets--default

I'll post my current algolia-related code in case it's of any help.

```javascript
<InstantSearch indexName={"image-consult-app"} searchClient={searchClient}>
                        <GetHitNumber />
                        <Configure 
                            hitsPerPage={20}
                            filters={"area:tokyo"}
                        />
                        <CustomSearchBox 
                            searchAsYouType={false}
                            defaultRefinement={searchWord}
                            urlWord={new URLSearchParams(props.location.search).get('query')}
                        />
                        <CustomStateResults />
                        <Hits hitComponent={hitComponent} />
                        <Pagination />
</InstantSearch>
```

**Comment by Haroenv** [comment]
Is this the documentation you're looking for? https://www.algolia.com/doc/api-reference/widgets/dynamic-facets/react/

**Reply by anoonoll** [reply]
Thanks for the quick answer!
Yes, I read that Doc, but I couldn't quite figure out how to use it... ;(

I tried to run it as follows, but it did not behave as I expected.

What's wrong with that?

```javascript
<InstantSearch indexName={"image-consult-app"} searchClient={searchClient}>
                       <DynamicWidgets>
  <RefinementList attribute="area:tokyo" />
</DynamicWidgets>
                        <GetHitNumber />
                        <Configure 
                            hitsPerPage={20}
                            filters={"area:tokyo"}
                        />
                        <CustomSearchBox 
                            searchAsYouType={false}
                            defaultRefinement={searchWord}
                            urlWord={new URLSearchParams(props.location.search).get('query')}
                        />
                        <CustomStateResults />
                        <Hits hitComponent={hitComponent} />
                        <Pagination />
</InstantSearch>
```

**Reply by Haroenv** [reply]
Could you make what you have in a sandbox? I've showed a demo dynamic widget here: https://codesandbox.io/s/empty-water-vonhw?file=/src/App.js

**Reply by anoonoll** [reply]
Thank you.
I've rewritten it!

**Reply by anoonoll** [reply]
I've solved myself! Thank you so much!

**Comment by anoonoll** [comment, marked as answer]
I solved myself by using Menu and the filters in Configure!
